From a634c65897c823993adc13ab857833c2b42e4f5f Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 30 Nov 2015 10:32:37 +0800
Subject: [PATCH] MLK-11917-2 media: output: pxp_v4l2: correct comparing
 unsigned to 0

Comparing unsigned integer with 0 is meaningless. So
remove this kind of integer comparing.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
(cherry picked from commit 71cc9d913122ec2e8fd17b9f6635b6e29b1df36a)
---
 drivers/media/platform/mxc/output/mxc_pxp_v4l2.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
index 3252f7b..22840da 100644
--- a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
+++ b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
@@ -418,7 +418,7 @@ static int pxp_enumoutput(struct file *file, void *fh,
 {
 	struct pxps *pxp = video_get_drvdata(video_devdata(file));
 
-	if ((o->index < 0) || (o->index > 1))
+	if (o->index > 1)
 		return -EINVAL;
 
 	memset(o, 0, sizeof(struct v4l2_output));
@@ -454,7 +454,7 @@ static int pxp_s_output(struct file *file, void *fh,
 	u32 size;
 	int ret, bpp;
 
-	if ((i < 0) || (i > 1))
+	if (i > 1)
 		return -EINVAL;
 
 	/* Output buffer is same format as fbdev */
@@ -493,9 +493,9 @@ static int pxp_enum_fmt_video_output(struct file *file, void *fh,
 				struct v4l2_fmtdesc *fmt)
 {
 	enum v4l2_buf_type type = fmt->type;
-	int index = fmt->index;
+	unsigned int index = fmt->index;
 
-	if ((fmt->index < 0) || (fmt->index >= ARRAY_SIZE(pxp_s0_formats)))
+	if (fmt->index >= ARRAY_SIZE(pxp_s0_formats))
 		return -EINVAL;
 
 	memset(fmt, 0, sizeof(struct v4l2_fmtdesc));
-- 
1.8.0

